### DISCUSSION TITLE
讨论当站点访问量过大时的应对方案

### DISCUSSION THREAD

**Discussion by MCSeekeri** [spencerwooo/onedrive-vercel-index#595] [Q&A] [answered]
我一直以为有了 Upstash，就没必要用其他数据库。
事实证明我错了。
![image](https://user-images.githubusercontent.com/20928094/162898153-7b95cc42-ae19-4ea2-9921-67a01f078176.png)
当站点访问量足够大时，就会超出 Upstash 的免费方案限制。导致整个站点进入不可访问状态。

目前我想到的几个应对方案：
1. 使用 Cloudflare Workers 做一层代理，利用 CF 强大的缓存能力防止后端数据库访问量过大。
这个方案简单易行，但问题是治标不治本，假如站点以大量小文件为主，CF 缓存的用处就不大，最终还是会超出限制。
Vercel 的[官方文档](https://vercel.com/support/articles/using-cloudflare-with-vercel?query=cloudflare#with-proxy)中也认为不应该给 Vercel 套一层 CDN。
2. 加入对 MongoDB 的支持
MongoDB 同样提供免费服务，而且额度似乎较大……
我翻过他们提供的文档，没有找到关于使用限额相关的信息，只提到了连接数限制和数据库大小之类。
所以加入对 MongoDB 的支持是可选项。
3. 抛弃数据库
馊主意。
4. 购买付费方案 
……没钱啊（叹气

**Comment by zkeq** [comment]
已经换上 `redislabs.com` 好香

------
------
------

我昨天已经成功将数据库转移到 replit.com，`replit` 有一个自带的数据库，采用的也是 `key` `value` 的形式，

简单做个 `http` 请求中转一下就可以为本项目所用

![image](https://user-images.githubusercontent.com/62864752/162906415-ae690a6c-4128-4549-8bef-301c8d060b8c.png)


我成功搭建好的 `示例站点` 是这个：https://tuostudy.com/ （速度虽然慢一点但是免费很香）

对应的 `replit` 仓库地址是这个 https://github.com/zkeq/backend_db_for_ovi

没什么技术含量，就是将 请求 `redis` 的文件修改了一下，好让它对接我搭建的 http 数据库..

https://github.com/zkeq/onedrive-vercel-index/blob/263062e72f81e739bdc588be2fe6662202668b07/utils/odAuthTokenStore.ts

这里只是提供一种可行的思路，可以试着部署一下，可能我会再优化一下来让他变得可以让更多人使用？

**Reply by code004Accepted** [reply]
原来在知乎看到的分享高考资料的就是您，幸会幸会😂😂😂

**Reply by kreva11** [reply]
没有下文了吗？

**Reply by zkeq** [reply]
> 

现在我使用的是 `Replit` + `Vercel` 反代

`vercel.json`
```json
{
    "version": 2,
    "routes": [
        {
            "src": "/(.*)",
            "dest": "https://域名/$1",
            "headers": {
                "cache-control": "ms-maxage=1000"
            }
        }
    ]
  }
```

我认为这是个很理想的解决方案

```json
https://tuostudy.com
```

![333](https://user-images.githubusercontent.com/62864752/166414167-c54272bc-a56c-486f-b33f-c6323afaf899.png)

**Reply by kreva11** [reply]
已经弃用该项目并转到cloudreve了，但还是非常感谢解答~，重新考虑了一下需求似乎cloudreve更适合我~

**Comment by MDsub** [comment, marked as answer]
redislabs.com 提供的30m存储限制的redis数据库，不介意没有tls的话可以试试

**Reply by zkeq** [reply]
换上了，真香，感谢大佬。

**Reply by MCSeekeri** [reply]
![Screenshot_2022-04-26-05-36-11-153_com android email](https://user-images.githubusercontent.com/20928094/165179348-f7c78a67-fb1a-464c-ab93-6049955f95f9.jpg)
完美。
这并没有解决我的问题……

**Comment by myl7** [comment]
Redis 目前仅用于[存储 OneDrive token](https://github.com/spencerwooo/onedrive-vercel-index/blob/f6ebe0f40e2f2ef84d0ba5b6a5c9ed2cf6ec2c5a/utils/odAuthTokenStore.ts) 且只有 set/get 操作，所以有需求的话可以自行对接其他的后端来存储，比如 Cloudflare KV
